packages/asc-scrapper/src/client.ts

```diff
@@ -0,0 +1,161 @@
+import { Axios, AxiosResponse } from 'axios';
+import {
+    BuildingsTableRow,
+    CardsTableRow,
+    ClassesTableRow,
+    ClassroomsTableRow,
+    DaysDefsTableRow,
+    DivisionsTableRow,
+    GroupsTableRow,
+    LessonsTableRow,
+    PeriodsTableRow,
+    StudentsTableRow,
+    SubjectsTableRow,
+    TeachersTableRow,
+    TermsDefsTableRow,
+    TimetableVersion,
+    TimetableVersionInfo,
+    TimetableVersionListRaw,
+    TimetableVersionRaw,
+    WeeksDefsTableRow,
+} from './types.js';
+import {
+    mapPeriodsTableRow,
+    mapDaysDefTableRow,
+    mapWeeksDefTableRow,
+    mapTermsDefTableRow,
+    mapClassroomsTableRow,
+    mapClassesTableRow,
+    mapSubjectsTableRow,
+    mapTeachersTableRow,
+    mapGroupsTableRow,
+    mapDivisionsTableRow,
+    mapStudentsTableRow,
+} from './mappers.js';
+import { getTableRowsById } from './utils.js';
+
+interface ServerRequestPayload {
+    __args: unknown[];
+    __gsh: '00000000';
+}
+
+interface ServerResponse<T> {
+    r: T;
+}
+
+export class Client {
+    private readonly axios: Axios;
+
+    constructor(axios: Axios, edupageInstanceId: string) {
+        this.axios = axios;
+        this.axios.defaults.baseURL = `https://${edupageInstanceId}.edupage.org/timetable/server`;
+    }
+
+    private async sendRequest<T>(controller: string, function_: string, args: unknown[]): Promise<T> {
+        const url = `/${controller}?__func=${function_}`;
+        const response = await this.axios.post<
+            ServerResponse<T>,
+            AxiosResponse<ServerResponse<T>, ServerRequestPayload>
+        >(url, {
+            __args: args,
+            __gsh: '00000000',
+        });
+        return response.data.r;
+    }
+
+    public async getTimetableVersionList(): Promise<TimetableVersionInfo[]> {
+        const response = await this.sendRequest<TimetableVersionListRaw>('ttviewer.js', 'getTTViewerData', [
+            null,
+            2099,
```

Comment:
Magiczna stała - warto dodać jakiś komentarz, bo nie wiem co ta liczba oznacza i dlaczego akurat taka

Reply:
To pole jest bardzo dziwne, tam powinno być id aktualnego roku szkolnego np. 2022 dla roku 2022/2023, ale API przyjmuje inne wartości.